phat/contracts/action_offchain_rollup/lib.rs

```diff
@@ -429,18 +429,37 @@ mod action_offchain_rollup {
         /// Processes a request with the the core js and returns the output.
         fn handle_request(&self, request: &[u8]) -> Result<(Vec<u8>, CodeHash)> {
             let Some(Core {
-                script,
                 settings,
                 code_hash,
+                script,
             }) = self.core.get()
             else {
                 error!("CoreNotConfigured");
                 return Err(Error::CoreNotConfigured);
             };
+            let compiled_core_script =
+                if let Some(compiled_script) = pink::ext().cache_get(&code_hash) {
+                    info!("Using cached core js");
+                    compiled_script
+                } else {
+                    info!("Compiling core js");
+                    let compiled_script = js::compile(&script)
+                        .or(Err(Error::JsError("Failed to compile core js".to_string())))?;
+                    let result = pink::ext().cache_set(&code_hash, &compiled_script);
```

Comment:
Note: This will require every rollup contract has sufficient stake. Otherwise they have no storage.

Reply:
> Note: This will require every rollup contract has sufficient stake. Otherwise they have no storage.

Yep, I mentioned it in the PR description. The alternative might be upload the compiled bytecode.
Maybe we can make a phat contract serves as a compiler and uploader.